src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/RecordTake.kt

```diff
@@ -131,15 +138,25 @@ class RecordTake(
             .insertForContent(take, content)
             .toCompletable()
 
-    fun record(project: Collection, chapter: Collection, content: Content): Completable {
+    fun record(project: Collection, chapter: Collection, content: Content): Single<Result> {
         return create(project, chapter, content)
                 .flatMap { take ->
                     launchPlugin
                             .launchRecorder(take.path)
-                            .toSingle { take }
-                }
-                .flatMapCompletable { take ->
-                    insert(take, content)
+                            .flatMap {
+                                when (it) {
+                                    LaunchPlugin.Result.SUCCESS -> {
+                                        if (take.path.readBytes().size == EMPTY_WAVE_FILE_SIZE) {
```

Comment:
this is very expensive- you're reading the entire file, which could be of unlimited size basically, into memory, just to check the size of the resulting array, rather than just asking the filesystem directly how big the file is.

just use take.path.length()

Reply:
even just opening a FileInputStream and trying to read 45 bytes before eof would be much faster than readbytes()